includes/wikia/traits/TitleTrait.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * A trait that let's you expand the base Title class
+ *
+ * @author Adam Karmiński <adamk@wikia-inc.com>
+ */
+
+trait TitleTrait {
+	// Required base-class methods
+	abstract function getNamespace();
+	abstract function getDBkey();
+
+	/**
+	 * Returns a list of pages that link to that Title. It is based on the method used in
+	 * Special:WhatLinksHere but returns raw results instead of a ready (and useless) HTML markup.
+	 *
+	 * @param int $level How deep should I go? 0 means to the bottom (which is limited to 2)
+	 * @return array
+	 */
+	public function getIndirectLinks( $level = 0 ) {
```

Comment:
It could, although refactoring Special:WLH is definitely not in the scope of this task.